src/state.py

```diff
@@ -327,3 +319,35 @@ def get_nginx_config(charm: ops.CharmBase) -> NginxConfig:
 
         configurations[config.hostname][config.path] = config
     return configurations
+
+
+def get_hostnames(charm: ops.CharmBase) -> tuple[Hostname, ...]:
+    """Get the hostnames from integration data.
+
+    Args:
+        charm: The charm to extract integration data from.
+
+    Raises:
+        IntegrationDataError: Invalid hostname in integration data.
+
+    Returns:
+        A list of hostnames.
+    """
+    hostnames: list[Hostname] = []
+    relations = charm.model.relations[CACHE_CONFIG_INTEGRATION_NAME]
```

Comment:
Does it make sense to use `charm.model.relations.get(CACHE_CONFIG_INTEGRATION_NAME)` in here?

Reply:
Good catch. Updated the code.